totp/src/main/java/com/okta/totp/otp_display/OtpDisplayScreen.kt

```diff
@@ -0,0 +1,195 @@
+        }
+    }
+}
+
+@Composable
+fun OtpCode(otpEntry: OtpEntry, modifier: Modifier = Modifier) {
+    Row(modifier = modifier) {
+        Column {
+            Text(
+                otpEntry.otpCode,
+                fontWeight = FontWeight.Bold,
+                color = Color.Blue,
+                fontSize = 16.sp
+            )
+            Text(stringResource(id = R.string.account_label, otpEntry.account))
+            otpEntry.issuer?.let {
+                Text(stringResource(id = R.string.issuer_label, otpEntry.issuer))
+            }
+        }
+        Spacer(modifier = Modifier.weight(1f))
+        Button(
+            modifier = Modifier
+                .align(Alignment.CenterVertically)
+                .padding(16.dp),
+            onClick = { otpEntry.delete() },
+        ) {
+            Icon(
+                imageVector = Icons.Filled.Delete,
+                contentDescription = stringResource(id = R.string.delete_button_description),
+            )
+        }
+    }
+}
+
+@Preview
+@Composable
+private fun PreviewOtpScreenList() {
+    val context = LocalContext.current
+    OtpScreenList(
+        listOf(
+            OtpEntry(
+                "123456789",
+                "accoun1t@gmail.com",
+                "issuer1",
+                onDeleteOtpEntry = {
+                    Toast.makeText(
+                        context,
+                        "Deleting entry 1",
+                        Toast.LENGTH_SHORT
+                    ).show()
+                }
+            ),
+            OtpEntry(
+                "321412341",
+                "account2@gmail.com",
+                "issuer2",
+                onDeleteOtpEntry = {
+                    Toast.makeText(
+                        context,
+                        "Deleting entry 2",
+                        Toast.LENGTH_SHORT
+                    ).show()
+                }
+            ),
+            OtpEntry(
+                "653156123",
+                "account3@gmail.com",
+                "issuer3",
+                onDeleteOtpEntry = {
+                    Toast.makeText(
+                        context,
+                        "Deleting entry 3",
+                        Toast.LENGTH_SHORT
+                    ).show()
+                }
+            ),
+        ),
+    )
+}
+
+data class OtpEntry(
+    val otpCode: String,
+    val account: String,
+    val issuer: String?,
+    private val onDeleteOtpEntry: () -> Unit,
+) {
+    fun delete() {
+        onDeleteOtpEntry()
+    }
+
+    override fun equals(other: Any?): Boolean {
+        if (other is OtpEntry) {
+            return other.otpCode == otpCode && other.account == account && other.issuer == issuer
+        }
+        return super.equals(other)
+    }
+
+    override fun hashCode(): Int {
+        var result = otpCode.hashCode()
+        result = 31 * result + account.hashCode()
+        result = 31 * result + (issuer?.hashCode() ?: 0)
+        return result
+    }
```

Comment:
This code was autogenerated by the IDE. I'm open to switching to using Objects.hash if that's preferable